framework/source/class/qx/ui/table/model/Filtered.js

```diff
@@ -99,18 +109,11 @@ qx.Class.define("qx.ui.table.model.Filtered",
      * @param the_needle {String} String to search
      * @param the_haystack {Array} Array, which should be searched
      * @return {Boolean} whether the search string was found.
+     * @deprecated {6.0} Use qx.lang.Array.contains instead
      */
     _js_in_array : function(the_needle, the_haystack)
     {
-      var the_hay = the_haystack.toString();
-
-      if (the_hay == '') {
-        return false;
-      }
-
-      var the_pattern = new RegExp(the_needle, 'g');
-      var matched = the_pattern.test(the_haystack);
-      return matched;
+      return qx.lang.Array.contains(the_haystack, the_needle);
```

Comment:
yes because you are using strings without active characters: ... try:

```javacript
new RegExp("a", "g").test(["aa", "ab", "ac"]) === true
new RegExp("c.+d", "g").test(["cad", "caedefdb", "cccccccdddddd"]) === true
new RegExp("[^ad]", "g").test(["a", "aaa", "d"]) === false
```